### PR TITLE
Unity weather parameters, weather HUD, and a visual effect for snow

### PR DESCRIPTION
**Pull request by ForrestTrepte**
This is a Unity implementation of weather analogous to AirSim's Unreal weather system. Currently only the snow weather type is implemented.

* Weather class stores global weather settings and applies them to instances of weather effects in the scene.
* AirSimGlobal singleton provides access to global Weather object.
* WeatherFX implements weather visuals surrounding a vehicle.
* HUD contains weather menu for enabling weather and setting snow quantity. Triggered by F10.
* Snow particle system.

This PR is a step towards #2395, but is not yet a complete implementation of weather in Unity. I haven't contributed to AirSim before, so please let me know if there's something that doesn't make sense or isn't following conventions.

### CODE REVIEW

**Comment by ForrestTrepte** [comment]
FYI, this is work from the Microsoft Hackathon 2020.

**Comment by ForrestTrepte** [comment]
I am unfamiliar with this build system. Can anyone help me understand why the CI build failed on Ubuntu 16.04? Perhaps related to this?
```
+echo '### Eigen is not installed. Please run setup.sh first.'
### Eigen is not installed. Please run setup.sh first.
+exit 1
```

**Comment by rajat2004** [comment]
It's just a rate limit on Travis which happens sometimes
```
+wget https://github.com/Microsoft/AirSim/releases/download/v1.2.0/car_assets.zip
--2020-08-04 01:31:41--  https://github.com/Microsoft/AirSim/releases/download/v1.2.0/car_assets.zip
Resolving github.com (github.com)... 140.82.113.3
Connecting to github.com (github.com)|140.82.113.3|:443... connected.
HTTP request sent, awaiting response... 429 too many requests
2020-08-04 01:31:41 ERROR 429: too many requests.
```
You could close and reopen the PR or rebase against master, or if anyone has permissions, they can restart the Travis build

**Comment by ForrestTrepte** [comment]
Thanks @rajat2004! I made an update and the checks passed this time.

**Review by zimmy87 — CHANGES_REQUESTED** [review]
Tested this locally and it works for me, however there's a few compile error and warnings we'd like to see addressed before merging this.

File: Unity/UnityDemo/Assets/AirSimAssets/Scripts/Weather/WeatherFX.cs [hunk 1]
```diff
@@ -0,0 +1,30 @@
+﻿using UnityEngine;
+
+/// <summary>
+/// Renders an instance of weather visual effects.
+/// </summary>
```
[the remark this answers is not in the document]
Comment (ForrestTrepte) [reply Unity/UnityDemo/Assets/AirSimAssets/Scripts/Weather/WeatherFX.cs]: moved

File: Unity/UnityDemo/Assets/AirSimAssets/Scripts/HUD/WeatherHUD.cs [hunk 2]
```diff
@@ -0,0 +1,49 @@
+﻿using System;
+using UnityEditor.VersionControl;
```
[the remark this answers is not in the document]
Comment (ForrestTrepte) [reply Unity/UnityDemo/Assets/AirSimAssets/Scripts/HUD/WeatherHUD.cs]: removed

File: Unity/UnityDemo/Assets/AirSimAssets/Scripts/HUD/WeatherHUDSlider.cs [hunk 3]
```diff
@@ -0,0 +1,56 @@
+﻿using System;
+using UnityEditor.UI;
```
[the remark this answers is not in the document]
Comment (ForrestTrepte) [reply Unity/UnityDemo/Assets/AirSimAssets/Scripts/HUD/WeatherHUDSlider.cs]: removed

File: Unity/UnityDemo/Assets/AirSimAssets/Scripts/HUD/WeatherHUD.cs [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+using UnityEngine.UI;
+
+namespace AirSimUnity {
+    /// <summary>
+    /// HUD class that manage weather-related UI elements.
+    /// </summary>
+    public class WeatherHUD : MonoBehaviour {
+        [SerializeField] private RectTransform rootPanel;
+        [SerializeField] private Toggle weatherEnabledToggle;
+        [SerializeField] private WeatherHUDSlider snowSlider;
```
[the remark this answers is not in the document]
Comment (ForrestTrepte) [reply Unity/UnityDemo/Assets/AirSimAssets/Scripts/HUD/WeatherHUD.cs]: fixed

File: Unity/UnityDemo/Assets/AirSimAssets/Scripts/Weather/Weather.cs [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+    /// </summary>
+    public class Weather : MonoBehaviour
+    {
+        public enum WeatherParamScalar {
+            Snow,
+            Count
+        }
+
+        [Tooltip("WeatherFX prefab that will be spawned as a child of each vehicle.")]
+        [SerializeField] private WeatherFX WeatherFXPrefab;
```
[the remark this answers is not in the document]
Comment (ForrestTrepte) [reply Unity/UnityDemo/Assets/AirSimAssets/Scripts/Weather/Weather.cs]: fixed

File: Unity/UnityDemo/Assets/AirSimAssets/Scripts/Weather/WeatherFX.cs [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+namespace AirSimUnity
+{
+    public class WeatherFX : MonoBehaviour
+    {
+        [SerializeField] private ParticleSystem snowParticleSystem;
```
[the remark this answers is not in the document]
Comment (ForrestTrepte) [reply Unity/UnityDemo/Assets/AirSimAssets/Scripts/Weather/WeatherFX.cs]: fixed

**Comment by ForrestTrepte** [comment]
Thanks for the review, @zimmy87. I made changes that I *think* are straightforward and resolve these issues. However, I wasn't able to test running them. It has been a long time since I worked on this and, currently, Unity crashes whenever I play the DroneDemo scene in the editor. It doesn't seem to be related to my changes. I crash with the original version of my PR branch, with these updates, or with the latest master branch. I'm not sure what has gone wrong and don't have time to investigate.

**Review by zimmy87 — CHANGES_REQUESTED** [review]
Tested this locally again and the latest version is working for me. There's one remaining item that I think is a a typo, once that's addressed, I think this will be ready for merging.

**Comment by ForrestTrepte** [comment]
Thanks for fixing my typo, testing, and merging @zimmy87!!

**Pull request merged**